File: backup/lircom.suspicious/advertise.php
<?php
$file = fopen("servers.txt", "r");
$servers = array();
while ($line = fgets($file)) {
	$values = explode("|", $line);
	$host = $values[0];
	$port = $values[1];
	$nick = $values[2];
	$key = implode("|", array($nick, $host, $port));
	$servers[$key] = $line;
}
fclose($file);

if ($_GET['nick']) {
	$nick = $_GET['nick'];
}
if ($_GET['port']) {
	$port = $_GET['port'];
}

$host = $_ENV['REMOTE_HOST'];
if (!$host) {
	$host = $_ENV['REMOTE_ADDR'];
}

if ($nick && $port) {
	$key = implode("|", array($nick, $host, $port));
	$tm = array();
	$tm = localtime(time(), true);

	$date = ($tm[tm_year]+1900)."-".($tm[tm_mon]+1)."-$tm[tm_mday] $tm[tm_hour]:$tm[tm_min]:$tm[tm_sec]";
	$servers[$key] = $host."|".$port."|".$nick."|".$date."\n";
}

$file = fopen("servers.txt", "w");
foreach ($servers AS $key => $value) {
	fwrite($file, $value);
	echo $value;
}
fclose($file);
?>

File: backup/lircom.2013/web/advertise.pl
#!E:/perl/bin/perl
#!/usr/bin/perl

use CGI;

print "Content-type: text/plain\n\n";

$query = new CGI;

if ($query->param('nick')) {
	$nick = $query->param('nick');
}
if ($query->param('port')) {
	$port = $query->param('port');
}

$host = $ENV{'REMOTE_HOST'};
if (!$host) {
	$host = $ENV{'REMOTE_ADDR'};
}

($sec,$min,$hour,$mday,$mon,$year,$wday,$yday) = gmtime(time);

$date = ($year+1900)."-".($mon+1)."-$mday $hour:$min:$sec";

if ($nick && $port) {
	open(SERVERS, ">>servers.txt");
	print SERVERS $host."|".$port."|".$nick."|".$date."\n";
	close(SERVERS);
}

open(SERVERS,"servers.txt");
while (<SERVERS>) {
	$line = $_;
	@line = split(/|/);
	$key = join("|", ($nick, $host, $port));
	$servers{$key} = $_;
}
close(SERVERS);

open(SERVERS, ">servers.txt");
foreach $key (keys %servers) {
	print SERVERS $servers{$key};
	print $servers{$key};
}
close(SERVERS);
